### PR TITLE
better color selector with popup

### LINKED ISSUE #1483: Better color selector in edit menu
**Is your feature request related to a problem? Please describe.**
It's very annoying when you can choose your color at top of edit menu - as it's also need in 'quick add' use case.

**Describe the solution you'd like**
Move color selector to the top as such:  

**Describe alternatives you've considered**
None.

**Additional context**
I have another stretch goal for how selector can work - best explained verbally.

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1483 

- Moved colorpicker to top of config modal
- Revamped colorpicker component. Added a compact, scrollable color palette popup to ColorPicker for improved user interface

### CODE REVIEW

**Comment by tijlleenders** [comment]
Looks good!
I'm sure we'll want some minor design changes - but this is a lot nicer in my opinion.

**Review by tijlleenders — DISMISSED** [review]

File: src/components/GoalsComponents/GoalConfigModal/ConfigGoal.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { useLocation } from "react-router-dom";
 import plingSound from "@assets/pling.mp3";
 
 import ZAccordion from "@src/common/Accordion";
-import ColorPalette from "@src/common/ColorPalette";
+import ColorPalette from "@src/common/ColorPicker";
```
Comment (Tushar-4781) [inline src/components/GoalsComponents/GoalConfigModal/ConfigGoal.tsx:12]: the component name must match with file name. can you please fix this :)

File: src/global.scss [hunk 2]
```diff
@@ -244,22 +244,42 @@ h6 {
     color: white;
   }
 }
-.colorPallette {
+.color-picker-container {
```
Comment (Tushar-4781) [inline src/global.scss:247]: global is getting crowder. can you create an index.scss file for this color picker component ?

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
@vinaybadgujar102 please make this 2 minor changes and we are gtg

**Review by Tushar-4781 — DISMISSED** [review]
@vinaybadgujar102 please resolve the conflicts and after that we can merge

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**